### PR TITLE
adds with react-i18next example

### PR DESCRIPTION
**Pull request by jamuhl**
Sample translating next.js apps using react-i18next:

- Includes routing incl. translation file splitting and lazy load on client
- Language detections on serverside
- Proper setting of initialProps to pass down initial translations for render take over in client
- ...

### CODE REVIEW

**Comment by timneutkens** [comment]
@jamuhl looks like travis failed on code style. Could you run `npm run lint -- --fix`? ❤️

**Review by timneutkens — APPROVED** [review]
Other than the codestyle this looks great!

File: examples/with-react-i18next/.babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["next/babel"],
```
Comment (timneutkens) [inline examples/with-react-i18next/.babelrc]: In this case `.babelrc` is obsolete. You can remove it.

Comment (jamuhl) [inline examples/with-react-i18next/.babelrc]: @timneutkens will check later today and provide updates...thank you for reviewing...

Comment (timneutkens) [inline examples/with-react-i18next/.babelrc]: Thanks! ❤️

Comment (jamuhl) [inline examples/with-react-i18next/.babelrc]: green now...sorry for not checking on doing PR.

**Comment by timneutkens** [comment]
Thanks!

**Review by timneutkens — APPROVED** [review]

**Comment by jamuhl** [comment]
@timneutkens awesome...thank you for the fast merge...

**Comment by kachkaev** [comment]
@jamuhl WDYT of converting all these lines into a HOC so that the pages did not look so scary? Is there any technical limitation to prevent that?

Here's roughly what I mean:

https://github.com/zeit/next.js/blob/e7d91bf69229c12fd0d6e63711d21d0b271706c4/examples/with-react-i18next/pages/index.js#L21-L30
↓
```js
// with-i18next.js

export const withI18next = (namespaces) => (Component) {
    const Extended = translate(namespaces, { i18n, wait: process.browser })(Component)

    Extended.getInitialProps = async ({ req }) => {
    if (req && !process.browser) return i18n.getInitialProps(req, ['home', 'common'])
    return {}
    }

    export default Extended
}
```

Then we should be able to:

```js
// index.js
export const Home = withI18n(['common', 'home'], ({ t, initialI18nStore }) => /**/ );

// page2.js
export const Page2 = withI18n(['page2', 'home'], ({ t, initialI18nStore }) => /**/ );
```

⚠️  I haven't checked this code, just mocked it up in the github comment 😉

**Comment by jamuhl** [comment]
@kachkaev looks nicer...my sample is just a show case...including an additional hoc to combine translateHOC and getInitialProps appending in one is for sure better...if you don't mind adding a PR would be awesome.

**Comment by kachkaev** [comment]
Thanks, I'll try! I just wanted to make sure I'm not missing anything important. This hoc (if I manage it to work, of course), may give `react-i18n` a couple of extra points compared to `react-intl` 😉

**Comment by kachkaev** [comment]
Also WDYT of adding `<Trans>` component in the PR? Could not find it in the example.

**Comment by jamuhl** [comment]
Feel free

**Comment by kachkaev** [comment]
@jamuhl see https://github.com/zeit/next.js/pull/3932

**Comment by kachkaev** [comment]
One more PR related to this example: https://github.com/zeit/next.js/pull/3944

**Comment by jamuhl** [comment]
@kachkaev awesome stuff...would you mind copying that stuff over to https://github.com/i18next/react-i18next/tree/master/example/nextjs

would be awesome to have your contribution backed into the repo - if you got no time i understand and will copy the changes over myself...

**Comment by kachkaev** [comment]
Yeah I was thinking of this and might find some time next week. Feel free to do the updates sooner yourself if you wish. There's actually one more thing that could be improved, see my question here: https://github.com/i18next/react-i18next/issues/397

Overall, I'm wondering if https://github.com/i18next/react-i18next/tree/master/example/nextjs could be just replaced with a folder saying _Please see https://github.com/zeit/next.js/tree/canary/examples/with-react-i18next_. IMO we can avoid the need of syncing this way and I don't see much downsides. WDYT?

**Comment by jamuhl** [comment]
@kachkaev agree it might be better not having to sync it...

**Comment by kachkaev** [comment]
Yeah, probably. Instead of syncing PRs between two repos, you can just store a copy of next.js in a safe just in case if it gets hijacked by Martians and goes out everyone's of control 😃  In this case you'll just replace README placeholder in https://github.com/i18next/react-i18next/tree/master/example/nextjs with the latest version of the code you've backed up.

Having `with-react-i18next` among next.js example is quite useful, because it creates a user funnel for you. I'm an example of those who have discovered your library by going to https://github.com/zeit/next.js/tree/canary/examples and then simply `cmd+F` `i18n`, `intl`, `trans`, `lang`, etc. 😄

**Comment by jamuhl** [comment]
@kachkaev will do an update to the readme on the sample over at react-i18next...later today.

**Pull request merged**